### PR TITLE
Use switch to load items.xml

### PR DESCRIPTION
**Pull request by Mkalo**
This is faster (not really that relevant as either way its fast) but I think it's easier to read and maintain. 

I want to know what you guys think about this, and also we could do something as mentioned in #1952 for some attributes. (just wanted to know if this will ever get merged before doing it.)

### CODE REVIEW

**Comment by A-Syntax** [comment]
You are still going to get a fatal error C1061 somewhere down the road, because all you did was band-aide the code, not much of a difference between using switch and if/else. 

As far as the changes go I fail to see how you simplified anything for the better. But what do I know you people don't even want me in your community (banning me for no reason at all). Whatever...

**Comment by Kamenuvol** [comment]
I think all these attributes should be in Lua side.
In C, there would be a single function to add any of these attributes in a map of attributes of that item; and other function to read its value.

File: src/items.h [hunk 1]
```diff
@@ -57,6 +57,220 @@ enum ItemTypes_t {
...
+	ITEM_PARSE_ELEMENTFIRE,
+	ITEM_PARSE_ELEMENTENERGY,
+	ITEM_PARSE_WALKSTACK,
+	ITEM_PARSE_BLOCKING,
+	ITEM_PARSE_ALLOWDISTREAD
+};
+
+const std::unordered_map<std::string, ItemParseAttributes_t> ItemParseAttributesMap = {
+	{"type",				ITEM_PARSE_TYPE},
+	{"description",				ITEM_PARSE_DESCRIPTION},
```
Comment (ranisalt) [inline src/items.h]: Either align every line or use a single space.

File: src/items.cpp [hunk 2]
```diff
@@ -372,502 +374,801 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
...
+						std::cout << "[Warning - Items::parseItemNode] Unknown field value: " << valueAttribute.as_string() << std::endl;
+					}
+
+					if (combatType != COMBAT_NONE) {
+						it.combatType = combatType;
+						it.conditionDamage.reset(conditionDamage);
+						uint32_t ticks = 0;
+						int32_t damage = 0;
+						int32_t start = 0;
+						int32_t count = 1;
```
Comment (ranisalt) [inline src/items.cpp]: All variables above should be inside for loop.

File: src/items.cpp [hunk 3]
```diff
@@ -372,502 +374,801 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
...
+					} else if (tmpStrValue == "axe") {
+						it.weaponType = WEAPON_AXE;
+					} else if (tmpStrValue == "shield") {
+						it.weaponType = WEAPON_SHIELD;
+					} else if (tmpStrValue == "distance") {
+						it.weaponType = WEAPON_DISTANCE;
+					} else if (tmpStrValue == "wand") {
+						it.weaponType = WEAPON_WAND;
+					} else if (tmpStrValue == "ammunition") {
+						it.weaponType = WEAPON_AMMO;
```
Comment (ranisalt) [inline src/items.cpp]: You should use a map instead of this if/elseif chain. It also applies to a lot of other places.

File: src/items.cpp [hunk 4]
```diff
@@ -372,514 +532,787 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
...
-				conditionDamage = new ConditionDamage(CONDITIONID_COMBAT, CONDITION_DROWN);
-				combatType = COMBAT_DROWNDAMAGE;
-			} else if (tmpStrValue == "physical") {
-				conditionDamage = new ConditionDamage(CONDITIONID_COMBAT, CONDITION_BLEEDING);
-				combatType = COMBAT_PHYSICALDAMAGE;
-			} else {
-				std::cout << "[Warning - Items::parseItemNode] Unknown field value: " << valueAttribute.as_string() << std::endl;
-			}
+		auto parseAttribute = ItemParseAttributesMap.find(tmpStrValue);
+		if(parseAttribute != ItemParseAttributesMap.end()) {
```
Comment (WibbenZ) [inline src/items.cpp]: if (

File: src/items.cpp [hunk 5]
```diff
@@ -29,6 +29,164 @@
...
+	{"femalesleeper", ITEM_PARSE_FEMALETRANSFORMTO},
+	{"transformto", ITEM_PARSE_TRANSFORMTO},
+	{"destroyto", ITEM_PARSE_DESTROYTO},
+	{"elementice", ITEM_PARSE_ELEMENTICE},
+	{"elementearth", ITEM_PARSE_ELEMENTEARTH},
+	{"elementfire", ITEM_PARSE_ELEMENTFIRE},
+	{"elementenergy", ITEM_PARSE_ELEMENTENERGY},
+	{"walkstack", ITEM_PARSE_WALKSTACK},
+	{"blocking", ITEM_PARSE_BLOCKING},
+	{"allowdistread", ITEM_PARSE_ALLOWDISTREAD}
```
Comment (WibbenZ) [inline src/items.cpp]: Add a comma here or remove from the the other maps

File: src/items.cpp [hunk 6]
```diff
@@ -372,514 +532,787 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
...
+				case ITEM_PARSE_BLOCKPROJECTILE: {
+					it.blockProjectile = valueAttribute.as_bool();
+					break;
+				}
+
+				case ITEM_PARSE_PICKUPABLE: {
+					it.allowPickupable = valueAttribute.as_bool();
+					break;
+				}
+
```
Comment (WibbenZ) [inline src/items.cpp]: Trailing

File: src/items.h [hunk 7]
```diff
@@ -57,6 +57,115 @@ enum ItemTypes_t {
...
+	ITEM_PARSE_FEMALETRANSFORMTO,
+	ITEM_PARSE_TRANSFORMTO,
+	ITEM_PARSE_DESTROYTO,
+	ITEM_PARSE_ELEMENTICE,
+	ITEM_PARSE_ELEMENTEARTH,
+	ITEM_PARSE_ELEMENTFIRE,
+	ITEM_PARSE_ELEMENTENERGY,
+	ITEM_PARSE_WALKSTACK,
+	ITEM_PARSE_BLOCKING,
+	ITEM_PARSE_ALLOWDISTREAD
```
Comment (WibbenZ) [inline src/items.h]: Same here, if you wanna add a comma or remove the others

File: src/items.cpp [hunk 8]
```diff
@@ -372,514 +532,788 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
...
+				case ITEM_PARSE_CONTAINERSIZE: {
+					it.maxItems = pugi::cast<uint16_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_FLUIDSOURCE: {
+					tmpStrValue = asLowerCaseString(valueAttribute.as_string());
+					if (tmpStrValue == "water") {
```
Comment (DSpeichert) [inline src/items.cpp]: why not make this a map?

File: src/items.cpp [hunk 9]
```diff
@@ -372,514 +532,788 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
...
+						it.weaponType = it2->second;
+					} else {
+						std::cout << "[Warning - Items::parseItemNode] Unknown weaponType: " << valueAttribute.as_string() << std::endl;
+					}
+					break;
+				}
+
+				case ITEM_PARSE_SLOTTYPE: {
+					tmpStrValue = asLowerCaseString(valueAttribute.as_string());
+					if (tmpStrValue == "head") {
```
Comment (DSpeichert) [inline src/items.cpp]: This could be a map as well

File: src/items.cpp (hunk 8 above)
Comment (ranisalt) [reply src/items.cpp]: Legacy code. Truly useful maps are a feature of C++11 and the codebase is probably older :P

**Comment by DSpeichert** [comment]
@Mkalo can you rebase this so we can merge?

**Pull request closed without merging**